src/noit_filters_rest.c

```diff
@@ -222,13 +246,18 @@ rest_set_filter(mtev_http_rest_closure_t *restc,
   snprintf(xpath, sizeof(xpath), "//filtersets%sfilterset[@name=\"%s\"]",
            pats[0], pats[1]);
   node = mtev_conf_get_section(NULL, xpath);
-  if(!node && noit_filter_exists(pats[1])) {
+  exists = noit_filter_get_seq(pats[1], &old_seq);
+  if(!node && exists == mtev_true) {
     /* It's someone else's */
     error_code = 403;
     goto error;
   }
 
-  if((newfilter = validate_filter_post(indoc, pats[1])) == NULL) goto error;
+  if((newfilter = validate_filter_post(indoc, pats[1], &seq)) == NULL) goto error;
+  if(exists && old_seq > seq) {
```

Comment:
I think this should be

if(exists && ((old_seq >= seq) || (seq == 0)) {

So that you can't replace a filter without bumping the sequence number up -- unless seq is just zero and stays zero.  I thought that was how checks worked.